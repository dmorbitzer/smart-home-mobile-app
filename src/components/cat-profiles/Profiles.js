import Icon from "@expo/vector-icons/MaterialCommunityIcons";
import { Box, ListItem } from "@react-native-material/core";

const size = 25;

export default function Profiles(props) {
  const navigateToScreen = (event, element) => {
    const screen =
      props.type === "profile"
        ? "Katzendetailansicht"
        : props.type === "feed"
        ? "FÃ¼tterung"
        : "";
    props.navigation.navigate(screen, {
      name: element.node.name,
      catId: element.node.id,
    });
  };
  const profilesList = props.data.map((element) => {
    return (
      <ListItem
        key={element.node.id}
        title={element.node.name}
        leading={<Icon name="cat" size={size} />}
        trailing={() => <Icon name="chevron-right" />}
        onPress={(event) => {
          navigateToScreen(event, element);
        }}
      />
    );
  });
  return <Box>{profilesList}</Box>;
}
