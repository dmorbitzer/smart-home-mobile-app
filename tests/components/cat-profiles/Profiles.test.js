import React from "react";
import renderer from "react-test-renderer";

import Profiles from "../../../src/components/cat-profiles/Profiles";
jest.useFakeTimers();

const localData = {
  data: {
    cats: {
      edges: [
        {
          node: {
            id: "/api/cats/8",
            name: "Peter",
            gender: {
              name: "male",
            },
            race: "Bengal",
            weight: 5.2,
          },
        },
        {
          node: {
            id: "/api/cats/9",
            name: "Susi",
            gender: {
              name: "female",
            },
            race: "Maine-Coon",
            weight: 8.5,
          },
        },
        {
          node: {
            id: "/api/cats/12",
            name: "Björn",
            gender: {
              name: "female",
            },
            race: "Blauhaar",
            weight: 0.5,
          },
        },
        {
          node: {
            id: "/api/cats/13",
            name: "Max",
            gender: {
              name: "female",
            },
            race: "Blauhaar",
            weight: 0.5,
          },
        },
        {
          node: {
            id: "/api/cats/229",
            name: "Hugo",
            gender: {
              name: "male",
            },
            race: "Grüngut",
            weight: 5.5,
          },
        },
      ],
    },
  },
};

const navigation = null;
test("renders CatProfiles correctly", () => {
  const tree = renderer
    .create(
      <Profiles data={localData.data.cats.edges} navigation={navigation} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
