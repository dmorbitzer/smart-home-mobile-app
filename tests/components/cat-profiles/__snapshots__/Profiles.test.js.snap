// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CatProfiles correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": undefined,
        "borderBottomColor": undefined,
        "borderBottomEndRadius": undefined,
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderBottomStartRadius": undefined,
        "borderBottomWidth": undefined,
        "borderColor": undefined,
        "borderEndColor": undefined,
        "borderEndWidth": undefined,
        "borderLeftColor": undefined,
        "borderLeftWidth": undefined,
        "borderRadius": undefined,
        "borderRightColor": undefined,
        "borderRightWidth": undefined,
        "borderStartColor": undefined,
        "borderStartWidth": undefined,
        "borderStyle": undefined,
        "borderTopColor": undefined,
        "borderTopEndRadius": undefined,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "borderTopStartRadius": undefined,
        "borderTopWidth": undefined,
        "borderWidth": undefined,
        "bottom": undefined,
        "end": undefined,
        "height": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": undefined,
        "overflow": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingEnd": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingStart": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "start": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "transparent",
          },
          {
            "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          },
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginStart": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 16,
                  "letterSpacing": 0.15,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#212121",
                },
              ]
            }
          >
            Peter
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginEnd": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#00000014",
            "height": 1,
            "marginEnd": undefined,
            "marginStart": 56,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "transparent",
          },
          {
            "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          },
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginStart": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 16,
                  "letterSpacing": 0.15,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#212121",
                },
              ]
            }
          >
            Susi
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginEnd": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#00000014",
            "height": 1,
            "marginEnd": undefined,
            "marginStart": 56,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "transparent",
          },
          {
            "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          },
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginStart": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 16,
                  "letterSpacing": 0.15,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#212121",
                },
              ]
            }
          >
            Björn
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginEnd": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#00000014",
            "height": 1,
            "marginEnd": undefined,
            "marginStart": 56,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "transparent",
          },
          {
            "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          },
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginStart": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 16,
                  "letterSpacing": 0.15,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#212121",
                },
              ]
            }
          >
            Max
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginEnd": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#00000014",
            "height": 1,
            "marginEnd": undefined,
            "marginStart": 56,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "transparent",
          },
          {
            "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          },
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginStart": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          {
            "flex": 1,
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 16,
                  "letterSpacing": 0.15,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#212121",
                },
              ]
            }
          >
            Hugo
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "marginEnd": 16,
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#00000014",
            "height": 1,
            "marginEnd": undefined,
            "marginStart": 56,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
